### PR TITLE
[e2e] Support dual-stack - utils

### PR DESCRIPTION
**Pull request by lzhecheng**
Signed-off-by: Zhecheng Li <zhechengli@microsoft.com>

#### What type of PR is this?
/kind testing

#### What this PR does / why we need it:
[e2e] Support dual-stack - utils
* Support dual-stack in e2e utils functions and methods
* Adjust related e2e test code
* Define dual-stack Service annotations
#### Which issue(s) this PR fixes:

Fixes #
Related #814

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: tests/e2e/utils/network_utils.go [hunk 1]
```diff
@@ -394,34 +422,37 @@ func WaitGetPIP(azureTestClient *AzureTestClient, ipName string) (pip aznetwork.
 	return
 }
 
-// SelectAvailablePrivateIP selects a private IP address in Azure subnet.
-func SelectAvailablePrivateIP(tc *AzureTestClient) (string, error) {
+// SelectAvailablePrivateIPs selects private IP addresses in Azure subnet.
+func SelectAvailablePrivateIPs(tc *AzureTestClient) ([]string, error) {
```
Comment (nilo19) [inline tests/e2e/utils/network_utils.go:426]: Can we unit test this one?

Comment (lzhecheng) [reply tests/e2e/utils/network_utils.go:486]: Yes. I turned "selectSubnets" and "findIPInSubnet" parts into 2 functions. Then I added a UT for the first one. As for the second one, the logic is the same as before and I think it is already clear. PTAL.

**Pull request merged**